scripts/code_style.py

```diff
@@ -137,6 +139,20 @@ def get_src_files(since: Optional[str]) -> List[str]:
                          is_file_autogenerated(filename))]
     return src_files
 
+def get_submodule_hash(commit: str, submodule: str) -> str:
+    cmd = ["git", "ls-tree", commit, submodule]
+    output = subprocess.check_output(cmd, universal_newlines=True)
+
+    # Need to extract submodule commit hash from the command output.
+    # Form is: <> commit <hash> <submodule>
+    hash_regex = r"\b[0-9a-f]{40}\b"
+    results = re.search(hash_regex, output)
+    if results:
+        return results.group(0)
+    else:
+        # Submodule did not exist for commit
+        return ""
```

Comment:
Returning the empty string here doesn't actually work. The caller creates a command line with an empty argument, i.e.
`["git", "-C", "framework", "log", "", "HEAD"`, which causes Git to complain
```
fatal: bad revision ''
```

You can try this with e.g. `code_style.py --since mbedtls-3.5.0` from a directory on `development` or 3.6.

It's not important to support this case. `code_style.py --since …` is mostly useful to compare the branch you're working on with the point where it forked, which is a recent commit. If we really wanted to support going back to when the submodule didn't exist, then we should treat all the files in the submodule as in scope. But, again, we don't actually care about this.